ext/mysql2/client.c

```diff
@@ -30,14 +30,21 @@ VALUE rb_hash_dup(VALUE other) {
     rb_raise(cMysql2Error, "MySQL client is not initialized"); \
   }
 
+#define REQUIRE_CLIENT(wrapper) \
```

Comment:
Are there cases where a `GET_CLIENT` would _not_ be followed by a `REQUIRE_CLIENT` ? If not, then this check for a valid client pointer could go into `GET_CLIENT`.